tools/configure_framework_commit.sh

```diff
@@ -24,13 +23,35 @@ else
   cd $FLUTTER_CLONE_REPO_PATH
 fi
 
-# Get the time of the youngest commit older than engine commit.
-# Git log uses commit date not the author date.
-# Before makes the comparison considering the timezone as well.
-COMMIT_NO=`git log --before="$LATEST_COMMIT_TIME_ENGINE" -n 1 | grep commit | cut -d ' ' -f2`
+if [[ $GIT_BRANCH =~ ^flutter-.*-candidate.*$ ]]
+then
+  # Coming from presubmit and assuming the correct branch has been already checked out.
+  COMMIT_NO=`git rev-parse HEAD`
+else
+  # Try to get release branch from the checkout.
+  RELEASE_BRANCH=`git branch -a --contains $ENGINE_COMMIT | grep 'flutter-.*-candidate.*' || true`
+  if [[ -z $RELEASE_BRANCH ]]
+  then
+    # If this is not a release branch commit get latest commit's time for the engine repo.
+    # Use date based on local time otherwise timezones might get mixed.
+    LATEST_COMMIT_TIME_ENGINE=`git log -1 --date=local --format="%cd"`
+    echo "Latest commit time on engine found as $LATEST_COMMIT_TIME_ENGINE"
+
+    # Get the time of the youngest commit older than engine commit.
+    # Git log uses commit date not the author date.
+    # Before makes the comparison considering the timezone as well.
+    COMMIT_NO=`git log --before="$LATEST_COMMIT_TIME_ENGINE" -n 1 | grep commit | cut -d ' ' -f2`
+  else
+    COMMIT_NO=`git rev-parse $RELEASE_BRANCH`
+  fi
+fi
+#else
+#  # If this is a release branch use the ToT commit of the same release branch as the engine.
+#  COMMIT_NO=`git rev-parse $RELEASE_BRANCH`
+#ifi
+
```

Comment:
Are there any plans to use the `find_commit.dart` that hixie made? - https://cs.opensource.google/flutter/flutter/+/master:dev/tools/bin/find_commit.dart;l=85?q=find_commit.dart&sq=&ss=flutter%2Fflutter

Reply:
We can move this new logic there. Find_commit suffers from the same issues in this script. On release branches the engine commits with the cherry picks are recent causing to select current versions of the framework.

Reply:
Actually using find_commit would be difficult in the short term. It relies on the base branched being checked out which is not the case yet. Adapting it will require a few days of work.